beacon-chain/blockchain/chain_info.go

```diff
@@ -118,6 +118,12 @@ type OptimisticModeFetcher interface {
 	IsOptimisticForRoot(ctx context.Context, root [32]byte) (bool, error)
 }
 
+// ExecutionPayloadFetcher defines a common interface that returns forkchoice
+// information about payload block hashes
+type ExecutionPayloadFetcher interface {
+	HashInForkchoice([32]byte) bool
```

Comment:
Note: both `FinalizedBlockHash` and `UnrealizedJustifiedPayloadBlockHash` are still in `FinalizationFetcher`, it may be better to move them here in `develop` later